### PR TITLE
Add parent_zone_name capability

### PR DESCRIPTION
**Pull request by SweetOps**
## What

* Add `parent_zone_name` capability

## Why

* User should be able to specify `parent_zone_id` or `parent_zone_name` to `tf_vanity`

### CODE REVIEW

**Comment by SweetOps** [comment]
screenshot 2017-09-19 17 38 08

File: main.tf [hunk 1]
```diff
@@ -84,9 +84,10 @@ data "aws_iam_policy_document" "default" {
...
 
 module "dns" {
-  source          = "git::https://github.com/cloudposse/tf_vanity.git?ref=tags/0.2.0"
-  aliases         = "${list(var.hostname)}"
-  parent_zone_id  = "${var.dns_zone_id}"
-  target_dns_name = "${aws_s3_bucket.default.website_domain}"
-  target_zone_id  = "${aws_s3_bucket.default.hosted_zone_id}"
+  source           = "git::https://github.com/cloudposse/tf_vanity.git?ref=tags/0.2.0"
+  aliases          = "${compact(list(signum(length(var.dns_zone_id)) == 1 || signum(length(var.dns_zone_name)) == 1 ? var.hostname : ""))}"
+  parent_zone_id   = "${length(var.dns_zone_id) > 0 ? var.dns_zone_id : ""}"
```
Comment (osterman) [inline main.tf]: This seems unnecessary. `dns_zone_id` is an empty string by default.

File: main.tf [hunk 2]
```diff
@@ -84,9 +84,10 @@ data "aws_iam_policy_document" "default" {
...
+  parent_zone_name = "${length(var.dns_zone_name) > 0 ? var.dns_zone_name : ""}"
```
Comment (osterman) [inline main.tf]: This seems unnecessary. `dns_zone_name` is an empty string by default.

File: README.md [hunk 3]
```diff
@@ -30,7 +46,8 @@ module "website" {
 | `tags`                              | `{}`           | Additional tags  (e.g. `map("BusinessUnit","XYZ")`                                                              | No       |
 | `delimiter`                         | `-`            | Delimiter to be used between `name`, `namespace`, `stage`, `arguments`, etc.                                    | No       |
 | `hostname`                          | `[]`           | Name of website bucket in `fqdn` format (e.g. `test.example.com`). IMPORTANT! Do not add trailing dot (`.`)     | Yes      |
-| `dns_zone_id`                       | ``             | ID of the hosted zone to contain the record                                                                     | Yes      |
+| `dns_zone_id`                       | ``             | ID of the hosted zone to contain the record                                                                     | No       |
```
Comment (osterman) [inline README.md]: Add:
"or specify `dns_zone_name` instead"

File: README.md [hunk 4]
```diff
@@ -30,7 +46,8 @@ module "website" {
...
+| `dns_zone_name`                     | ``             | Name of the hosted zone to contain the record                                                                   | No       |
```
Comment (osterman) [inline README.md]: "or specify `dns_zone_id` instead"

File: README.md [hunk 5]
```diff
@@ -8,17 +8,33 @@ Terraform Module for Creating S3 backed Websites
...
+* Required one of the `dns_zone_id` or `dns_zone_name`
+
+```terraform
+module "website_with_cname" {
+  source      = "git::https://github.com/cloudposse/tf_s3_website.git?ref=master"
+  namespace   = "${var.namespace}"
+  stage       = "${var.stage}"
+  name        = "${var.name}"
+  hostname    = "${var.hostname}"
   dns_zone_id = "${var.dns_zone_id}"
```
Comment (osterman) [inline README.md]: wrong var

File: README.md [hunk 6]
```diff
@@ -8,17 +8,33 @@ Terraform Module for Creating S3 backed Websites
...
   source      = "git::https://github.com/cloudposse/tf_s3_website.git?ref=master"
   namespace   = "${var.namespace}"
   stage       = "${var.stage}"
   name        = "${var.name}"
   hostname    = "${var.hostname}"
-  dns_zone_id = "${var.dns_zone_id}"
 }
 ```
 
+#### Create s3 website bucket with associated CNAME of your (sub)domain hosted in Route53
```
Comment (osterman) [inline README.md]: It's actually not a `CNAME` but `ALIAS` of type `A`

https://github.com/cloudposse/tf_vanity/blob/master/main.tf#L26

Comment (osterman) [reply README.md]: Let's just write 
````
#### Create S3 Website Bucket with Route53 DNS
````

**Comment by comeanother** [comment]
Tested with `parent_zone_name`.
screenshot from 2017-09-20 11-17-55

screenshot from 2017-09-20 11-18-17

**Comment by comeanother** [comment]
Tested with `parent_zone_id`
screenshot from 2017-09-20 11-28-35

screenshot from 2017-09-20 11-28-56

**Pull request merged**